login.js

```diff
@@ -0,0 +1,72 @@
+const emailInputElement = document.querySelector('#email');
+const pwInputElement = document.querySelector('#pw');
+const errorMessage = document.querySelector('.error-message');
+const pwErrorMessage = document.querySelector('.error-message-pw');
+
+function emailForm(e) {
```

Comment:
P1:
변수명 함수명은 좀 더 유의미하고 어떤 동작을 하는 지 알 수 있는 이름이면 좋겠습니다.
해당 함수에서 인자 `e`는 사용되지 않으니 지워주시는게 더 좋겠습니다.

```suggestion
function checkEmailValidation() {
```

Reply:
P3:
아래처럼도 작성 가능합니다.

```js
const checkEmailFormatValid = (value: string) => {
  return !(value.includes("@") && value.includes("."))
}

function checkEmailValidation(e) {
  const input = e.currentTarget;
  const value = input.value.trim();

  if (!value) {
    emailInputElement.classList.add('error');
    errorMessage.textContent ="이메일을 입력해주세요."
    errorMessage.style.display = 'block';
    return ;
  }

  if (checkEmailFormatValid(value)) {
    emailInputElement.classList.add('error');
    errorMessage.textContent ="잘못된 이메일 형식입니다."
    errorMessage.style.display = 'block';
    return;
  }

  emailInputElement.classList.remove('error');
  errorMessage.style.display = 'none';
}
```

지금은 에러종류가 많지 않지만 나중에 에러종류가 많은 경우가 생긴다면 
아래처럼 작성할 수도 있습니다.

```js
const getEmailError = (value: string): {type: EMAIL_ERROR_TYPE} => {
  /** 이메일 input의 value를 받아 해당하는 에러 타입을 반환하는 함수 */
}


function checkEmailValidation(e) {
  const input = e.currentTarget;
  const errorType = getEmailError(input.value.trim());

  if (errorType) {
    emailInputElement.classList.add('error');
    errorMessage.textContent =ErrorMessage[errorType]
    errorMessage.style.display = 'block';
    return ;
  }

  emailInputElement.classList.remove('error');
  errorMessage.style.display = 'none';
}
```